src/trnpy/trnsys/lib.py

```diff
@@ -153,80 +149,64 @@ def set_input_value(self, unit: int, input_number: int, value: float) -> int:
 
 
 class LoadedTrnsysLib(TrnsysLib):
-    """Represents a TRNSYS library loaded in memory."""
-
-    def __init__(self, lib_path: Path):
+    """Represents a loaded TRNSYS library ready to run a simulation."""
+
+    def __init__(
+        self,
+        trnsys_dir: Path,
+        input_file: Path,
+        user_type_libs: Optional[List[Path]] = None,
+    ):
         """Initialize a LoadedTrnsysLib object.
 
         Raises:
-            DuplicateLibraryError: If the file at `lib_path` is already in use.
+            UnsupportedOperatingSystem: If the OS is supported by TRNSYS.
+            DuplicateLibraryError: If the libs in `trnsys_dir` are already in use.
             OSError: If an error occurs when loading the library.
         """
-        track_lib_path(lib_path)
-
-        self.lib = ct.CDLL(str(lib_path), ct.RTLD_GLOBAL)
-        self.lib_path = lib_path
-
-        # Define the function signatures
-        self.lib.apiSetDirectories.argtypes = [
-            ct.c_char_p,  # root dir
-            ct.c_char_p,  # exe dir
-            ct.c_char_p,  # user lib dir
-            ct.POINTER(ct.c_int),  # error code (by reference)
-        ]
-        self.lib.apiLoadInputFile.argtypes = [
-            ct.c_char_p,  # input file
-            ct.c_char_p,  # semicolon-separated list of type lib files
-            ct.POINTER(ct.c_int),  # error code (by reference)
-        ]
-        self.lib.apiStepForward.restype = ct.c_bool
-        self.lib.apiStepForward.argtypes = [
-            ct.c_int,  # number of steps
-            ct.POINTER(ct.c_int),  # error code (by reference)
-        ]
-        self.lib.apiGetCurrentTime.restype = ct.c_double
-        self.lib.apiGetCurrentTime.argtypes = [
-            ct.POINTER(ct.c_int),  # error code (by reference)
-        ]
-        self.lib.apiGetOutputValue.restype = ct.c_double
-        self.lib.apiGetOutputValue.argtypes = [
-            ct.c_int,  # unit number
-            ct.c_int,  # output number
-            ct.POINTER(ct.c_int),  # error code (by reference)
-        ]
-        self.lib.apiSetInputValue.argtypes = [
-            ct.c_int,  # unit number
-            ct.c_int,  # input number
-            ct.c_double,  # value to set
-            ct.POINTER(ct.c_int),  # error code (by reference)
-        ]
-
-    def set_directories(self, dirs: TrnsysDirectories) -> int:
-        """Set the TRNSYS directories in the library.
-
-        Refer to the documentation of `TrnsysLib.set_directories` for more details.
-        """
+        lib = _load_api_lib(trnsys_dir)
+
         error = ct.c_int(0)
-        self.lib.apiSetDirectories(
-            str(dirs.root).encode(),
-            str(dirs.exe).encode(),
-            str(dirs.user_lib).encode(),
+        lib.apiSetDirectories(
+            str(trnsys_dir).encode(),
+            str(trnsys_dir).encode(),
+            str(trnsys_dir).encode(),
```

Comment:
TRNSYS v18 makes assumptions about its directory structure.  I don't understand exactly why these different directories are needed (and my hope is that someday they go away completely in the kernel code) but for now we need to set these three directories in the kernel:

```
  !> `rootDir` - Main installation directory, e.g. "C:\Trnsys17"
  !> `exeDir` - Directory with EXE programs, e.g. "C:\Trnsys17\Exe"
  !> `userLibDir` - Directory with user DLLs, e.g.   "C:\Trnsys17\UserLib\ReleaseDLLs"
```

For now with TrnPy we're requiring that all three are the same directory, and users need to know to stick all lib files and resource files in that one directory (with the exception of user Type libs, which can live anywhere).  When TESS decide how to package v19 we may need to change this requirement, but until then this seemed like the easiest thing to do.

Related: I think the concept of a `userLibDir` will go away at the kernel level, since TrnPy or trnsys-exe will be responsible for gathering up any user-provided libs and passing the full paths to the kernel with the deck to load.

Reply:
I see. This probably lends itself well to using forced kwargs then so that it's clear what each of this is when calling. Basically the signature of `lib.apiSetDirectories` would be something like:

```python
def apiSetDirectories(*, rootDir, exeDir, userLibDir):
    ...
```